File: docs/api/typescript/run_workflow.ts
/**
 * Execute a workflow via the engine (TypeScript)
 * Run: npx tsx docs/api/typescript/run_workflow.ts
 */
const ENGINE = "http://localhost:8000";

async function runWorkflow() {
  const resp = await fetch(`${ENGINE}/api/orchestrator/execute`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({
      name: "TS Test Run",
      workflowId: `wf-ts-${Date.now()}`,
      goalFileUrl: "",
      steps: [{
        stepId: "s1", name: "Execute",
        taskDescription: process.argv[2] || "Write a hello world Python script.",
        modelSource: "local",
        localModel: { engine: "ollama", model: "llama3.2:3b" },
        temperature: 0.7, maxTokens: 1024, connectionType: "sequential",
      }],
    }),
  });
  const data = await resp.json();
  console.log(`✓ Run: ${data.runId}, Status: ${data.status}`);

  // Poll
  for (let i = 0; i < 20; i++) {
    await new Promise(r => setTimeout(r, 5000));
    const st = await fetch(`${ENGINE}/api/orchestrator/status`).then(r => r.json());
    console.log(`  [${(i+1)*5}s] active=${st.active_runs}`);
    if (st.active_runs === 0) { console.log("✓ Done"); break; }
  }
}

runWorkflow().catch(console.error);

File: docs/api/workflows/run_workflow.py
"""Trigger workflow execution via engine."""
import requests, json, sys, time

ENGINE = "http://localhost:8000"
FRONTEND = "http://localhost:3000"

wf_name = sys.argv[1] if len(sys.argv) > 1 else "Test Workflow"
goal = sys.argv[2] if len(sys.argv) > 2 else "Generate a simple Python script."

# Upload goal file
with open("/tmp/_goal.md", "w") as f:
    f.write(goal)
upload = requests.post(f"{ENGINE}/api/orchestrator/upload", files={"files": open("/tmp/_goal.md", "rb")})
goal_url = upload.json().get("files", [{}])[0].get("url", "")
print(f"Goal uploaded: {goal_url}")

# Execute
body = {
    "name": wf_name,
    "workflowId": f"wf-api-{int(time.time())}",
    "goalFileUrl": goal_url,
    "steps": [{
        "stepId": "s1", "name": "Execute",
        "taskDescription": goal,
        "modelSource": "local",
        "localModel": {"engine": "ollama", "model": "llama3.2:3b"},
        "temperature": 0.7, "maxTokens": 2048, "connectionType": "sequential",
    }],
}

resp = requests.post(f"{ENGINE}/api/orchestrator/execute", json=body)
data = resp.json()
run_id = data.get("runId", "?")
print(f"✓ Run started: {run_id}")

# Poll
for i in range(30):
    time.sleep(5)
    st = requests.get(f"{ENGINE}/api/orchestrator/status").json()
    active = st.get("active_runs", 0)
    print(f"  [{(i+1)*5}s] active={active}")
    if active == 0:
        print("✓ Completed")
        break
